sphinx/writers/latex.py

```diff
@@ -792,26 +792,24 @@ def generate_numfig_format(self, builder):
             ret.append('\\def\\fnum@figure{%s}\n' %
                        text_type(figure[0]).strip().translate(tex_escape_map))
         else:
-            definition = text_type(figure[0]).strip().translate(tex_escape_map)
+            definition = escape_abbr(text_type(figure[0]).translate(tex_escape_map))
             ret.append(self.babel_renewcommand('\\figurename', definition))
-            if figure[1]:
-                ret.append('\\makeatletter\n')
-                ret.append('\\def\\fnum@figure{\\figurename\\thefigure%s}\n' %
-                           text_type(figure[1]).strip().translate(tex_escape_map))
-                ret.append('\\makeatother\n')
+            ret.append('\\makeatletter\n')
+            ret.append('\\def\\fnum@figure{\\figurename \\thefigure{}%s}\n' %
```

Comment:
actually there is no reason why I inserted a space after `\\figurename` here, it is not significant and swallowed by TeX, independently of actual contents of `\\figurename` macro. I will remove that.